### ISSUE TITLE
Re-organise the `common` package

### ISSUE THREAD

**Issue by cyberbikepunk** [os-data/eu-structural-funds#302]
@akariv 

### Current modules

This issue is not urgent. It's about improving the overall coherence of the `common` package. As for now, there are 4 modules: 

1. `bootstrap` is home to the `Source` class and `click` command line functions that are not just about bootstrapping
2. `utilities` is a mixture of functions concerned with fetching specs, together with the `process` function and other helper functions that are either unused or would be better included in the modules where they are used
3. `config` holds configuration parameters and specs nomenclature
4. `metrics` holds a tiny bit of code (the model) that should be refactored into the `Source` class; the rest is obsolete

### Suggested changes

I would split the code into the following modules:

1. `source`: the `Source` class
2. `cli`: the `click` command line functions
3. `config`: strictly configuration parameters (no specs)
4. `specs`: everything that has to do with fetching specs information. I would assign the results of all the functions directly inside the module, e.g. instead of importing `get_fiscal_field_name` in other modules, the result `fiscal_fields_name` would be imported.
5. `metrics`: should be dumped

Finally, I suggest we move the `utilities.process` function to `datapackage-pipelines`. Some of the generalizable functionality of the current  `bootstrap` module could be moved too at one point. But I would let the dust settle a little longer before we do this.

**Comment by cyberbikepunk** [comment]
# Brain dump

The above comments are not very clear and perhaps too specific. Here's how I would rephrase it...

## Move code

Which functionality do we want to move up to `datapackage-pipelines`? 

### CLI tools

A large part of the `Source` class and most of the CLI could potentially be moved. I've been using them on a regular basis. I would qualify them as very useful. We'd have to

- separate what is 
- there's some overlap with `dpp` though.

### Processor boilerplate

The processor boilerplate could be moved too. We could start by moving a slightly improved version of `utilities.process` (perhaps just adding `stats` support) and work on the `Process` class in the meantime (see [#15](https://github.com/frictionlessdata/datapackage-pipelines/issues/15))

## Refactor the processor sub-package

Here's me reflecting on the current mess.

### Agree on some ground rules

- Schema should always be in sync with the data
- Processor behavior should depend as little as possible specific assumptions and as much as possible on information contained in the schema
- (*corollary*) Processors should be commutative whenever possible 
- Parameter override rules should be set in stone 

__note__:  there are two types of parameters: data related (e.g. field mapping instructions) and non data related (e.g. the verbosity or size of the sample dump in the log)

__suggestion__: 

### Merge and deprecate processors 

- Merge that belong together. One typical example: `reshape_data` and `load_fiscal_datapackage` are two facets of the same coin: the former mutates the data, the latter the datapackage.

- Get rid of all obsolete module: 

###



As stated in the previous commnent, this

1. Re-organise the `common` package as suggested in the previous comment
4. 

## .... comment in construction.... bear with me

**Comment by akariv** [comment]
(The functionality within `Source` was already assimilated in the `datapackage-pipeline` framework)
